### PR TITLE
Ravinder | Development | Purchase Order v 1.2

### PR DESCRIPTION
**Pull request by ravinderwalkover**
Ravinder | Development | Purchase Order v 1.2

### CODE REVIEW

File: apps/web-giddh/src/app/purchase/create-purchase-order/create-purchase-order.component.ts [hunk 1]
```diff
@@ -2810,4 +2818,75 @@ export class CreatePurchaseOrderComponent implements OnInit, OnDestroy {
...
+
+    /**
+     * This will check if we need to show pipe symbol
+     *
+     * @param {number} loop
+     * @returns {boolean}
+     * @memberof CreatePurchaseOrderComponent
+     */
+    public checkIfPipeSymbolRequired(loop: number): boolean {
+        if (loop < (this.purchaseOrders.length - 1)) {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/purchase/create-purchase-order/create-purchase-order.component.ts:2857]: Consider using just
```suggestion
        return (loop < (this.purchaseOrders.length - 1));
```

File: apps/web-giddh/src/app/purchase/purchase-order-preview/purchase-order-preview.component.scss [hunk 2]
```diff
@@ -976,3 +976,25 @@ table.tax-table.item {
...
+.loader:before {
+    display: block;
+    position: absolute;
+    z-index: 1000000;
+    content: "";
+    left: 0px;
+    top: 0px;
+    width: 200px;
+    height: 4px;
+    background-color: #5b64c9;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/purchase/purchase-order-preview/purchase-order-preview.component.scss:998]: Use color variables

File: apps/web-giddh/src/app/purchase/purchase-order/purchase-order.component.ts [hunk 3]
```diff
@@ -596,4 +644,69 @@ export class PurchaseOrderComponent implements OnInit, OnDestroy {
...
+    /**
+     * This will validate bulk update form and will update fields
+     *
+     * @memberof PurchaseOrderComponent
+     */
+    public validateBulkUpdateFields(): void {
+        let isValid = true;
+
+        if (this.bulkUpdateGetParams.action) {
+            if (this.bulkUpdateGetParams.action === 'purchasedate') {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/purchase/purchase-order/purchase-order.component.ts:670]: Consider adding these actions to an enum

File: apps/web-giddh/src/app/purchase/purchase-record/component/purchase-record.component.html [hunk 4]
```diff
@@ -1,3 +1,19 @@
-<div class="mt-1">
-    <app-invoice-preview [selectedVoucher]="'purchase'"></app-invoice-preview>
-</div>
+<div _ngcontent-xlm-c43="" class="invoce-controll">
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/purchase/purchase-record/component/purchase-record.component.html:1]: Incorrect spelling of control

File: apps/web-giddh/src/app/purchase/purchase-record/component/purchase-record.component.ts [hunk 5]
```diff
@@ -2,18 +2,27 @@ import { Component, OnInit } from '@angular/core';
...
 
 @Component({
     styleUrls: [`./purchase-record.component.scss`],
     templateUrl: './purchase-record.component.html'
 })
+
 export class PurchaseRecordComponent implements OnInit {
+    public activeTab: string = 'order';
+    /* Observable to unsubscribe all the store listeners to avoid memory leaks */
+    private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/purchase/purchase-record/component/purchase-record.component.ts:18]: Do add ngOnDestroy and emit this Subject

File: apps/web-giddh/src/app/purchase/purchase-record/component/purchase-record.component.ts [hunk 6]
```diff
@@ -23,4 +32,8 @@ export class PurchaseRecordComponent implements OnInit {
     public ngOnInit(): void {
         this.store.dispatch(this.generalAction.setAppTitle('/pages/purchase-management/purchase'));
     }
+
+    public onTabChanged(tabName: string): void {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/purchase/purchase-record/component/purchase-record.component.ts:39]: Missing JS doc

File: apps/web-giddh/src/app/services/purchase-order.service.ts [hunk 7]
```diff
@@ -175,4 +175,38 @@ export class PurchaseOrderService {
...
+    }
+
+    /**
+     * This will get the PDF
+     *
+     * @param {*} getRequestObject
+     * @returns {Observable<BaseResponse<any, any>>}
+     * @memberof PurchaseOrderService
+     */
+    public getPdf(getRequestObject: any): any {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/services/purchase-order.service.ts:204]: Avoid incorrect return type:
```suggestion
    public getPdf(getRequestObject: any): Observable<BaseResponse<any, any>> {
```

File: apps/web-giddh/src/app/services/purchase-record.service.ts [hunk 8]
```diff
@@ -99,4 +99,56 @@ export class PurchaseRecordService {
...
+    }
+
+    /**
+     * This will get the PDF
+     *
+     * @param {*} requestObject
+     * @returns {*}
+     * @memberof PurchaseRecordService
+     */
+    public getPdf(requestObject: any): any {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/services/purchase-record.service.ts:145]: Avoid incorrect return type in JS doc
```suggestion
    public getPdf(requestObject: any): Observable<BaseResponse<any, any>> {
```

File: apps/web-giddh/src/app/shared/purchase-order-preview/purchase-order-preview.component.scss [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+.loader:before {
+    display: block;
+    position: absolute;
+    z-index: 1000000;
+    content: "";
+    left: 0px;
+    top: 0px;
+    width: 200px;
+    height: 4px;
+    background-color: #5b64c9;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/shared/purchase-order-preview/purchase-order-preview.component.scss:20]: Use color variables

File: apps/web-giddh/src/assets/css/style-2.scss [hunk 10]
```diff
@@ -619,6 +619,12 @@ input {
     color: #262626 !important;
 }
 
+.invoce-controll .nav > li > a:hover{
```
Comment (prateek-giddh) [inline apps/web-giddh/src/assets/css/style-2.scss:622]: Avoid incorrect spelling `invoice-control`

File: apps/web-giddh/src/app/invoice/preview/invoice.preview.component.ts [hunk 11]
```diff
@@ -280,18 +295,23 @@ export class InvoicePreviewComponent implements OnInit, OnChanges, OnDestroy {
         this.exportedInvoiceBase64res$ = this.store.select(p => p.invoice.exportInvoicebase64Data).pipe(takeUntil(this.destroyed$));
         this.universalDate$ = this.store.select(p => p.session.applicationDate).pipe(takeUntil(this.destroyed$));
         this.voucherDetails$ = this.store.pipe(select(s => s.receipt.voucher), takeUntil(this.destroyed$));
+        this.companyName$ = this.store.pipe(select(state => state.session.companyUniqueName), takeUntil(this.destroyed$));
         //this._invoiceService.getTotalAndBalDue();
     }
-    openModal(template: TemplateRef<any>) {
+
+    public openModal(template: TemplateRef<any>): void {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/preview/invoice.preview.component.ts:302]: Missing JS doc

File: apps/web-giddh/src/app/invoice/preview/invoice.preview.component.ts [hunk 12]
```diff
@@ -88,6 +90,9 @@ export class InvoicePreviewComponent implements OnInit, OnChanges, OnDestroy {
     @ViewChild('advanceSearchComponent', { read: InvoiceAdvanceSearchComponent }) public advanceSearchComponent: InvoiceAdvanceSearchComponent;
     @Input() public selectedVoucher: VoucherTypeEnum = VoucherTypeEnum.sales;
     @ViewChild(InvoicePaymentModelComponent) public invoicePaymentModelComponent: InvoicePaymentModelComponent;
+    @Input() public refreshPurchaseBill: boolean = false;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/preview/invoice.preview.component.ts:93]: Missing JS doc

**Review by prateek-giddh — CHANGES_REQUESTED** [review]

**Review by aditi-tiwari17 — APPROVED** [review]

**Review by prateek-giddh — APPROVED** [review]

**Pull request merged**